### ISSUE TITLE
fastai2.1.5 TextDataLoaders.from_df resulting in unexpected results

### ISSUE THREAD

**Issue by sayanbanerjee32** [fastai/fastai#2969]
Please confirm you have the latest versions of fastai, fastcore, fastscript, and nbdev prior to reporting a bug (delete one): YES 

**Describe the bug**
While using TextDataLoaders.from_df from latest fastai version 2.1.5, the labels are getting incorrectly assigned against the texts. However using the same steps in v2.1.4 does not have the issue.

**To Reproduce**
1. install fastai2.1.5 and run the following code

  from sklearn.datasets import fetch_20newsgroups
  import pandas as pd
  
  def twenty_newsgroup_to_df(data_set = 'train'):
      newsgroups_data = fetch_20newsgroups(subset=data_set, 
                                          remove=('headers', 'footers',
                                                  'quotes'))

      df = pd.DataFrame([newsgroups_data.data, 
                       newsgroups_data.target.tolist()]).T
      df.columns = ['text', 'target']

      targets = pd.DataFrame(newsgroups_data.target_names)
      targets.columns = ['title']

      data_set = pd.merge(df, targets, left_on='target', right_index=True)
      return data_set

  train_data_df = twenty_newsgroup_to_df('train')

  dls = TextDataLoaders.from_df(train_data_df, text_col='text', label_col='title', seed=1)
  dls.show_batch(max_n=5)

2. crosscheck the mapping of text to category shown in last statement with original data frame

**Expected behavior**
It was expected that the text to category mapping will remain unchanged after TextDataLoaders.from_df

**Error with full stack trace**
There was no error.
Place between these lines with triple backticks:

```
```

**Additional context**
I was trying this google colab after upgrading fastai to the latest version.

**Comment by nagads** [comment]
@sayanbanerjee32 hi, as dataloader.show_batch changes the order of dataset, and does not have export functionality, could you please let me know how you compared original data frame with dataloader. thanks
